### PR TITLE
fix(sql): don't classify objects with permissive __getattr__ as connections

### LINKED ISSUE #10213: marimo notebook hangs when using pytorch-ignite metraics
### Describe the bug

Instantiating a any `ignite.metrics.Metric` instance it just hangs that particular cell executing it. I tried and tested this with `ignite.metrics.Loss` and `ignite.metrics.RunningAverage`. It it just remains hung. It just never terminates, the hourglass keeps spinning...

Image

Further when you try to stop the cell by using the interrupt button, it doesn't get interrupted, here is the log message when clicking that button:

Image

I also checked for the specific codebases of [PyTorch Ignite](https://github.com/pytorch/ignite), more specifically the source codes of `ignite.metrics.Loss` and `ignite.metrics.RunningAverage` but couldn't find anything problematic there. At least I didn't see any problem in their `__init__` method. Please look into this. Or suggest a fix. The issue is same when running locally as well as on _molab_

### Will you submit a PR?

- [ ] Yes

### Environment

<details>

```
OS: Windows 11
Python: 3.14.0 (Miniconda)
marimo: 0.23.14
```

</details>

### Code to reproduce

```python
# In any marimo code cell run the following
import marimo as mo
import torch
from ignite.metrics import Loss, RunningAverage
loss = Loss(torch.nn.CrossEntropyLoss()) # or any other metric practically
```

### PR DESCRIPTION
**Pull request by Jayashanker-Padishala**
**This pull request was authored by a coding agent.** (Per AGENTS.md: authored by Claude Code under the direction of @Jayashanker-Padishala, who takes responsibility for it. Marked as draft per policy.)

Fixes #10213

## 📝 Summary

Instantiating any `ignite.metrics.Metric` hung the executing cell forever. Root cause (found via faulthandler stack dump of the hung kernel):

1. ignite metrics implement `__getattr__` to return a lazy `MetricsLambda` factory for **any** attribute name — and each access runs `traceback.format_stack()` (ignite `metric.py:797`), so every attribute probe is also slow.
2. That makes every `callable(getattr(var, method, None))` duck-typing check succeed, so the post-execution `_broadcast_data_source_connection` hook classified the metric as an **ADBC connection**.
3. `AdbcConnectionCatalog.get_databases` → `adbc_get_objects(...).read_all().to_pylist()` then walked into the lazy object graph and never produced real data — the cell appears hung and is not interruptible.

## 🔧 Fix

Add `fabricates_attributes()` (in `_sql/engines/types.py`): probe one attribute that no real connection defines; if the object returns something for it, its duck-typing answers are meaningless and `is_compatible` rejects it. Applied to both `AdbcDBAPIEngine` and `DBAPIEngine` (the two engines relying on pure `getattr` duck-typing — the others use isinstance checks).

Deliberately **not** `inspect.getattr_static`: genuine connection-pool proxies legitimately forward `__getattr__` to an underlying connection and would regress; they pass the probe (the probe attribute doesn't exist on the real connection either).

This generalizes the existing ibis `Deferred` special-case (#7791) — any permissive-`__getattr__` object (lazy proxies, mocks) is now handled, not just known names.

## 🧪 Verification

- Reproduced the hang in the interactive kernel (`executing_kernel` fixture + `from ignite.metrics import Loss; metric = Loss(nn.MSELoss())`): watchdog fired at 30s pre-fix; post-fix the same cell completes in ~4s with `metric` defined.
- New tests: `test_is_compatible_rejects_fabricated_attributes` (dbapi), `test_adbc_is_compatible_rejects_fabricated_attributes` (adbc), `test_fabricated_attributes_not_treated_as_engine` (get_engines).
- `pytest tests/_sql/`: 246 passed. `ruff check` + `ruff format --check` clean on all changed files.

### CODE REVIEW

**Comment by Jayashanker-Padishala** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Light2Dark** [comment]
Thanks @Jayashanker-Padishala! I made a similar fix here with some additions https://github.com/marimo-team/marimo/pull/10301. But thank you for checking, appreciate it.
